gridcomps/History3G/HistoryCollectionGridComp_private.F90

```diff
@@ -137,4 +143,75 @@ function create_output_bundle(hconfig, import_state, rc) result(bundle)
       _RETURN(_SUCCESS)
    end function create_output_bundle
 
+   function create_output_alarm(clock, hconfig, rc) result(alarm)
+      type(ESMF_Alarm) :: alarm
+      type(ESMF_Clock), intent(inout) :: clock
+      type(ESMF_HConfig), intent(in) :: hconfig
+      integer, intent(out), optional :: rc
+
+      integer :: status
+      type(ESMF_HConfig) :: time_hconfig
+      type(ESMF_TimeInterval) :: time_interval
+      character(len=:), allocatable :: iso_time
+      type(ESMF_Time) :: first_ring_time, currTime, startTime
+      integer :: int_time, yy, mm, dd, m, h, s
+      logical :: has_ref_time, has_frequency
+
+      call ESMF_ClockGet(clock, currTime=currTime, timeStep=time_interval, startTime = startTime, _RC)
+      int_time = 0 
+
+      time_hconfig = ESMF_HConfigCreateAt(hconfig, keyString='time_spec', _RC)
+
+      has_frequency = ESMF_HConfigIsDefined(time_hconfig, keyString='frequency', _RC)
+      if (has_frequency) then
+         time_interval = hconfig_to_esmf_timeinterval(time_hconfig, 'frequency', _RC)
+      end if
```

Comment:
Seems like we need a `time_interval` even if there is not a 'frequency'.  Should default to heartbeat.

Reply:
@clune, look more carefully, I do, if the frequency is not in the hconfig, time_interval has already been set to the timeStep of the clock.

Reply:
Ah.  Was looking too hard for `time_interval = ...`.

Generally speaking, I really prefer to put those 2 pieces of logic closer together so that the 2 ways of defining the variable are clear to the reader.   At the very least,  `int_time = 0` should be moved down to the next code chunk where it is overridden in the hconfig.      Then you could still move the `ESMF_ClockGet` call down to just before the `has_frequency = ...` line.

Then, as far as I can tell, startTime is not used and could be eliminated.

Reply:
@tclune startTime is used later, line 183